### PR TITLE
I love Fnaf!!!!

### PR DESCRIPTION
**Pull request by fnaffan1987harharhar**
## About The Pull Request

I have been told that someone REFUSED to add fnaf sprites to this Space Station 31 thing.
As my duty of the carrier of the torch of FNAF, the only one personally blessed by Scott Cauthorn to carry on his legacy I have made it my sacred duty to spread his genius to every corner of the world I can.
Digitally because they said I was a flight risk.
BVut I will do my best to carry on that tourch!!

## Why It's Good For The Game

Uhhhh???? It's FNAF?!!!!
What other reason do you need?

Once my mandated three hours of time with my annoying bastard of a son is over I'll be har-har-harppy (get it?) to show some in-game screenshots!

Sprites from Virgo who took up the torch before anyone else.

## Changelog

:cl: Greg Hanson, 46, Fnaf Fan.
add: Fnaf has been added to the vendors. Scott is pleased for now. I can rest.
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -31,3 +31,34 @@
 	icon_state = "vest_worn"
 	inhand_icon_state = null
 	armor_type = /datum/armor/suit_armor
+
+obj/item/clothing/suit/costume/cardborg/fnaf //Virgo Sprites
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/armor.dm:35]: Link the PR that added them to Virgo, or the codebase itself

File: modular_zubbers/code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -31,3 +31,34 @@
...
+obj/item/clothing/suit/costume/cardborg/fnaf/bunny
+	name = "animatronic bunny suit"
+	desc = "Also comes in Yellow!"
+	icon_state = "bonniesuit"
+
+obj/item/clothing/suit/costume/cardborg/fnaf/foxy
+	name = "animatronic fox suit"
+	desc = "YARR!"
+	icon_state = "foxysuit"
+
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/armor.dm:64]:
```suggestion
/obj/item/clothing/suit/costume/cardborg/fnaf //Virgo Sprites
	name = "freddy fuckballs"
	desc = "is that the freddy fazbear????"
	icon = 'modular_zubbers/icons/mob/clothing/suits/armor.dmi'
	worn_icon = 'modular_zubbers/icons/mob/clothing/suits/armor.dmi'
	icon_state = "freddysuit"
	inhand_icon_state = null
	lefthand_file = null
	righthand_file = null

/obj/item/clothing/suit/costume/cardborg/fnaf/bear
	name = "animatronic bear suit"
	desc = "O, cholera."
	icon_state = "freddysuit"

/obj/item/clothing/suit/costume/cardborg/fnaf/bird
	name = "animatronic bird suit"
	desc = "Let's Eat!"
	icon_state = "chicasuit"

/obj/item/clothing/suit/costume/cardborg/fnaf/bunny
	name = "animatronic bunny suit"
	desc = "Also comes in Yellow!"
	icon_state = "bonniesuit"

/obj/item/clothing/suit/costume/cardborg/fnaf/foxy
	name = "animatronic fox suit"
	desc = "YARR!"
	icon_state = "foxysuit"

```

File: modular_zubbers/code/modules/vending/wardrobe.dm [hunk 3]
```diff
@@ -8,3 +8,15 @@
...
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/foxy = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/foxy = 1,
+	)
+
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/vending/wardrobe.dm:22]: This probably won't work with how the autodrobe uses product categories

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by Majkl-J** [comment]
Also I personally hate this but eh what can one do.

File: modular_zubbers/code/modules/vending/wardrobe.dm (hunk 3 above)
Comment (fnaffan1987harharhar) [reply modular_zubbers/code/modules/vending/wardrobe.dm:22]: It didn't!!! I am unsure how else to do this. Maybe a crate. But I feel that such a lowley object is unworthy of holding the glory of fnaf.

**Comment by fnaffan1987harharhar** [comment]
Scott is angry.

File: modular_zubbers/code/modules/vending/vending.dm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+	for (var/obj/item/clothing/item in premium)
+		if(premium[item] < MINIMUM_CLOTHING_STOCK && allow_increase(item))
+			premium[item] = MINIMUM_CLOTHING_STOCK
+
+	QDEL_NULL(zubbers_products)
+	QDEL_NULL(zubbers_product_categories)
+	QDEL_NULL(zubbers_premium)
+	QDEL_NULL(zubbers_contraband)
+	return ..()
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/vending/vending.dm:59]: You just put in all the code I was putting off for weeks. Ok you may have your masks

File: modular_zubbers/code/modules/vending/vending.dm [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	// Don't touch bodyarmour!
+	if(ispath(clothing, /obj/item/clothing/suit/armor))
+		return FALSE
+	// Don't touch protective helmets, like riot helmets!
+	if(ispath(clothing, /obj/item/clothing/head/helmet))
+		return FALSE
+	// Ignore all gloves, because it's almost impossible to check what they do...
+	if(ispath(clothing, /obj/item/clothing/gloves))
+		return FALSE
+	return TRUE
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/vending/vending.dm:80]: Already defined in skyrat code, no need to do it again

```suggestion
```

**Comment by fnaffan1987harharhar** [comment]
Scott please. Listen. I have no one. I just have you. This is an offering to you do not abandon me. I have not abandoned me yet all the ordeals you have sent me I beg you please

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by Majkl-J** [comment]
Ok let me just stop you here before you get out of scope, add the modular vending stuff to the one vendor.

I will touch up the rest of the current modular overrides right after https://github.com/Bubberstation/Bubberstation/pull/838 is done, and move it to a standalone module or something.

As much as I love the work you're doing, this is getting out of scope too fast

**Comment by fnaffan1987harharhar** [comment]
> Ok let me just stop you here before you get out of scope, add the modular vending stuff to the one vendor.
> 
> I will touch up the rest of the current modular overrides right after #838 is done, and move it to a standalone module or something.
> 
> As much as I love the work you're doing, this is getting out of scope too fast

As you demand...

File: modular_zubbers/code/modules/vending/wardrobe.dm [hunk 6]
```diff
@@ -1,10 +1,25 @@
...
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/foxy = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/foxy = 1,
+			),
+		),
```
Comment (fnaffan1987harharhar) [inline modular_zubbers/code/modules/vending/wardrobe.dm:25]:
```suggestion
		),
		
		
```

**Comment by fnaffan1987harharhar** [comment]
scott i have given all i have for you, I have bled, I have secluded myself except for when mortal laws dictate I must, grant me this, I beg you

**Comment by fnaffan1987harharhar** [comment]
@Majkl-J Approve the workflow. Please, I must know the cause of the errors, if any...

**Comment by fnaffan1987harharhar** [comment]
image
Something seems to be getting angry at the name in the dome.

**Comment by Cyprex** [comment]
Uhhhhhhhh copyright. Also this feels LRP but idk

**Comment by fnaffan1987harharhar** [comment]
SCOTT HAS CAME DOWN AND BLESSED ME

HE HAS OPENED MY BLINDED EYES YET AGAIN AND I HAVE SOLVED IT

**Comment by fnaffan1987harharhar** [comment]
image
Scott. I thank you, I am humble, as your servant for always for the fact that you are so kind, so willing to allow me to spread your word to the masses. You have saved me. I thankyou.

File: modular_zubbers/code/modules/clothing/suits/armor.dm [hunk 7]
```diff
@@ -31,3 +31,34 @@
 	icon_state = "vest_worn"
 	inhand_icon_state = null
 	armor_type = /datum/armor/suit_armor
+
+/obj/item/clothing/suit/costume/cardborg/fnaf //Virgo Sprites, https://github.com/VOREStation/VOREStation
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/armor.dm:35]:
```suggestion
// FNAF COSTUMES BELOW
// Credit:
// Virgo Sprites, from: https://github.com/VOREStation/VOREStation

/obj/item/clothing/suit/costume/cardborg/fnaf 
```

File: modular_zubbers/modules/clothing/head/helmet.dm [hunk 8]
```diff
@@ -27,3 +27,37 @@
 	name = "magnificent pinwheel hat"
 	desc = "The strongest possible pinwheel pinwheel hat. Such is fate that the silliest things in the world are also the most beautiful; others may not see the shine in you, but the magnificent pinwheel hat does. It appreciates you for who you are and what you've done. It feels alive, and makes you feel alive too. You see the totality of existence reflected in the golden shimmer of the pin." //Does literally nothing more than the regular pinwheel hat. Just for emphasis.
 	icon_state = "pinwheel_gold"
+
+/obj/item/clothing/head/costume/cardborg/fnaf
```
Comment (Majkl-J) [inline modular_zubbers/modules/clothing/head/helmet.dm:31]:
```suggestion
// FNAF MASKS BELOW
// Credit:
// Virgo Sprites, from: https://github.com/VOREStation/VOREStation

/obj/item/clothing/head/costume/cardborg/fnaf
```

File: modular_zubbers/code/modules/vending/wardrobe.dm [hunk 9]
```diff
@@ -8,3 +8,21 @@
...
+			"icon" = "mask",
+			"products" = list(
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bear = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bird = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/bunny = 1,
+		/obj/item/clothing/suit/costume/cardborg/fnaf/foxy = 1,
+		/obj/item/clothing/head/costume/cardborg/fnaf/foxy = 1,
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/vending/wardrobe.dm:24]:
```suggestion
				/obj/item/clothing/suit/costume/cardborg/fnaf/bear = 1,
				/obj/item/clothing/head/costume/cardborg/fnaf/bear = 1,
				/obj/item/clothing/suit/costume/cardborg/fnaf/bird = 1,
				/obj/item/clothing/head/costume/cardborg/fnaf/bird = 1,
				/obj/item/clothing/suit/costume/cardborg/fnaf/bunny = 1,
				/obj/item/clothing/head/costume/cardborg/fnaf/bunny = 1,
				/obj/item/clothing/suit/costume/cardborg/fnaf/foxy = 1,
				/obj/item/clothing/head/costume/cardborg/fnaf/foxy = 1,
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Otherwise looks good from the code/credit perspective.

**Comment by projectkepler-ru** [comment]
> Uhhhhhhhh copyright. Also this feels LRP but idk

yeah

**Comment by magatsuchi** [comment]
going to be honest with you, this is a low-effort meme pr and im going to treat it like such

from the shitpost-esque clothing descriptions, to the reason for making this which would be accepted absolutely nowhere

no

**Pull request closed without merging**